src/notation/internal/notationinteraction.cpp

```diff
@@ -3779,10 +3782,37 @@ void NotationInteraction::addBoxes(BoxType boxType, int count, AddBoxesTarget ta
             if (target == AddBoxesTarget::BeforeSelection) {
                 if (beforeBoxIndex < 0 || itemMeasureIndex < beforeBoxIndex) {
                     beforeBoxIndex = itemMeasureIndex;
+                    selectedItem = item;
+                    selectedItemMeasure = itemMeasure;
                 }
             } else {
                 if (itemMeasureIndex + 1 > beforeBoxIndex) {
                     beforeBoxIndex = itemMeasureIndex + 1;
+                    selectedItem = item;
+                    selectedItemMeasure = itemMeasure;
+                }
+            }
+        }
+
+        // special cases for "between measures elements"
+        if (selectedItem && selectedItemMeasure) { // null check
+            ElementType selectedItemType = selectedItem->type();
+            if (selectedItemType == ElementType::CLEF || selectedItemType == ElementType::BAR_LINE
+                || selectedItemType == ElementType::TIMESIG || selectedItemType == ElementType::KEYSIG) {
+                Fraction itemTick = selectedItem->tick();
+                Fraction measureTick = selectedItemMeasure->tick();
+                Fraction measureLastTick = measureTick + selectedItemMeasure->ticks();
+
+                if (itemTick == measureTick) {
+                    if (target == AddBoxesTarget::AfterSelection) {
+                        beforeBoxIndex -= 1;
```

Comment:
I found one odd case. When you create a clef and use the Properties panel to put it after the barline,
<img width="483" alt="Scherm­afbeelding 2024-07-18 om 20 08 36" src="https://github.com/user-attachments/assets/39854af4-b2a5-48a7-b876-793f2d2a502f">
and then insert a measure after it, the measure goes before it:
<img width="483" alt="Scherm­afbeelding 2024-07-18 om 20 11 22" src="https://github.com/user-attachments/assets/d6ba3c7f-0155-411b-a214-f93aa87e2658">
Perhaps the problem is that the clef should not be moved in this case.

Reply:
Thank You, Ill take a look.

Reply:
Ok, I think #23676 is an issue (I reported it separately, as I think, it doesnt belong here), will try to fix it too.